src/client/datascience/interactive-common/interactiveBase.ts

```diff
@@ -1230,6 +1230,8 @@ export abstract class InteractiveBase extends WebviewPanelHost<IInteractiveWindo
                                 this.owningResource
                             )
                             .then(noop, noop);
+                    } else {
```

Comment:
Sholdn't we re-throw the error here.
At this point, `askForLocalKernel` has returned `undefined` meaning, the user didn't select a kernel and for some reason kernel startutp failed.

I.e. prefer to throw the same  exception (bubble it all the way up), instead of returning `undefined` & then trying to figure out why we have `undefined` & the like.

Reply:
I had that originally. Throwing here makes the error just show up but then it tries again. Breaking here stops the cycle of asking for a new kernel.

Reply:
But what part causes the retry, is it inside this loop?
Why not break out of this loop & then throw the error.
I'm not really a fan of `undefined`, its akin to C++, where we return a bogus value & then call some global function to find out what failed (get error number).
I'd prefer to just bomb out with the error information.

Reply:
Else we end up with checking for `undefined` when it should have ideally just failed with an exceptoni.

Reply:
No it's not inside this loop. It's outside that retries if there's an exception.

Reply:
I could change the outside to respond to a canceled exception with a different response.

Reply:
Althought it's difficult to get in this situation. I have to create a kernel spec that is invalid but is still python.

Reply:
I can't repro this state anymore. I had a python interpreter that we weren't generating the kernelspec correctly for. Now that I fixed that I can't get this specific thing to happen anymore.

Switching kernels already behaved correctly if there was an error. It's only on starting the notebook that this would happen.

Reply:
I'd agree with Don that the undefined makes this function a bit messier with returning. Per your last comment do we still need the undefined state? Sounds like the state might not be a thing anymore. Not the biggest deal, just do agree that "createNotebook => Valid notebook or throw" is better than "createNotebook => Valid notebook, or throw, or ?."